File: 2952-minimum-number-of-coins-to-be-added/2952-minimum-number-of-coins-to-be-added.java
class Solution {
    public int minimumAddedCoins(int[] coins, int target) {
        Arrays.sort(coins);
        int ans=0;
        int curr=1;
        for(int x:coins)
        {
            while(curr<x)
            {
                curr*=2;
                ans++;
            }
            curr+=x;
            if(curr>target)
                break;
        }
        while(curr<=target)
        {
            curr*=2;
            ans++;
        }
        return ans;
    }
}

File: 2952-minimum-number-of-coins-to-be-added/2952-minimum-number-of-coins-to-be-added.cpp
class Solution {
public:
    int minimumAddedCoins(vector<int>& coins, int target) {
        sort(coins.begin(),coins.end());
        long long int ans=0,x=1;
        for(auto i:coins){
            while(i>x){
                ans++;
                x=x*2;
            }
            x+=i;
            if(x>target)
                break;
        }
        while(x<=target){
            ans++;
            x=x*2;
        }
        return ans;
    }
};